roles/test_operator/tasks/main.yml

```diff
@@ -0,0 +1,248 @@
+  when: not cifmw_test_operator_dry_run | bool
+
+- name: Wait for the Tempest job to be Ready
+  kubernetes.core.k8s_info:
+    kubeconfig: "{{ cifmw_openshift_kubeconfig }}"
+    api_key: "{{ cifmw_openshift_token | default(omit) }}"
+    context: "{{ cifmw_openshift_context | default(omit) }}"
+    namespace: "{{ cifmw_test_operator_namespace }}"
+    kind: Job
+    name: "{{ cifmw_test_operator_tempest_name }}"
+    wait: true
+  when: not cifmw_test_operator_dry_run | bool
+
+- name: Wait for the Tempest job to be Completed
+  kubernetes.core.k8s_info:
+    kubeconfig: "{{ cifmw_openshift_kubeconfig }}"
+    api_key: "{{ cifmw_openshift_token | default(omit) }}"
+    context: "{{ cifmw_openshift_context | default(omit) }}"
+    namespace: "{{ cifmw_test_operator_namespace }}"
+    kind: Job
+    name: "{{ cifmw_test_operator_tempest_name }}"
+    wait: true
+    wait_timeout: "{{ cifmw_test_operator_timeout }}"
+    wait_condition:
+      type: Complete
+      status: true
+  when: not cifmw_test_operator_dry_run | bool
+
+- name: Start test-operator-logs-pod
+  when: not cifmw_test_operator_dry_run | bool
+  kubernetes.core.k8s:
+    kubeconfig: "{{ cifmw_openshift_kubeconfig }}"
+    api_key: "{{ cifmw_openshift_token | default(omit)}}"
+    context: "{{ cifmw_openshift_context | default(omit)}}"
+    state: present
+    definition:
+      apiVersion: v1
+      kind: Pod
+      metadata:
+        name: test-operator-logs-pod
+        namespace: "{{ cifmw_test_operator_namespace }}"
+      spec:
+        containers:
+          - name: test-operator-logs-container
+            image: "{{ cifmw_test_operator_logs_image }}"
+            command: ["/bin/sh", "-c", "--"]
+            args: ["while true; do sleep 30; done;"]
+            volumeMounts:
+              - name: logs-volume
+                mountPath: /mnt
+        volumes:
+          - name: logs-volume
+            persistentVolumeClaim:
+              claimName: "{{ cifmw_test_operator_tempest_config.metadata.name }}"
+
+- name: Ensure that the test-operator-logs-pod is Running
+  kubernetes.core.k8s_info:
+    kubeconfig: "{{ cifmw_openshift_kubeconfig }}"
+    api_key: "{{ cifmw_openshift_token | default(omit) }}"
+    context: "{{ cifmw_openshift_context | default(omit) }}"
+    namespace: "{{ cifmw_test_operator_namespace }}"
+    kind: Pod
+    name: test-operator-logs-pod
+    wait: true
+  register: logs_pod
+  until: logs_pod.resources[0].status.phase == "Running"
+  delay: 10
+  retries: 20
+  when: not cifmw_test_operator_dry_run | bool
+
+- name: Get logs from test-operator-logs-pod
+  environment:
+    KUBECONFIG: "{{ cifmw_openshift_kubeconfig }}"
+    PATH: "{{ cifmw_path }}"
+  ansible.builtin.shell: >
+    oc cp -n {{ cifmw_test_operator_namespace }} openstack/test-operator-logs-pod:mnt/
+    {{ cifmw_test_operator_artifacts_basedir }}
+  when: not cifmw_test_operator_dry_run | bool
+
+- name: Get list of all pods
```

Comment:
filtering isn't allowed here ?

Reply:
I think it isn't. Or at least I wasn't able to figure out how. I need to find a partial match because the name of the pod is: `tempest-tests-[random_suffix]` and partial match is probably not supported.

Reply:
Isn't the pods has some sot of annotation or lable what you can use ?